aggregators/aggregator.go

```diff
@@ -603,23 +603,134 @@ func (a *Aggregator) processHarvest(
 	hs := harvestStats{
 		eventsTotal:            cm.EventsTotal,
 		youngestEventTimestamp: modelpb.ToTime(cm.YoungestEventTimestamp),
-		servicesOverflowed:     hllSketchEstimate(cm.OverflowServicesEstimator),
 	}
-	addOverflow := func(o *aggregationpb.Overflow) {
+
+	a.getAndLogOverflowStats(&hs, cm, aggIvl)
+
+	if err := a.cfg.Processor(ctx, cmk, cm, aggIvl); err != nil {
+		return harvestStats{}, fmt.Errorf("failed to process combined metrics ID %s: %w", cmk.ID, err)
+	}
+	return hs, nil
+}
+
+func (a *Aggregator) getAndLogOverflowStats(hs *harvestStats, cm *aggregationpb.CombinedMetrics, aggIvl time.Duration) {
+	hs.servicesOverflowed = hllSketchEstimate(cm.OverflowServicesEstimator)
+
+	logFunc := func(msg string, fields ...zap.Field) {
+		if a.cfg.OverflowLogging.Func != nil &&
+			(a.cfg.OverflowLogging.AggregationInterval == 0 || a.cfg.OverflowLogging.AggregationInterval == aggIvl) {
+			a.cfg.OverflowLogging.Func(msg, fields...)
+		}
+	}
+
+	// TODO(carsonip): Update this log message when global labels implementation changes
+	logFunc(fmt.Sprintf(""+
+		"Service limit reached, new metric documents will be grouped under a dedicated "+
+		"overflow bucket identified by service name '%s'. "+
+		"If you are sending global labels that are user-specific (e.g. client IP), it may cause "+
+		"high cardinality and lead to exhaustion of services.", overflowBucketName),
+		zap.Duration("aggregation_interval_ns", aggIvl),
```

Comment:
Should we make this `aggregation_interval` and use `formatDuration(aggIvl)`, to match the OTel metric dimension attribute we use?

Reply:
Hmm I see now we're using `aggregation_interval_ns` in logs elsewhere, so I guess leave it for now, but we should probably bring them together at some point.